src/dbnode/namespace/schema.go

```diff
@@ -278,20 +297,43 @@ func marshalFileDescriptors(fdList []*desc.FileDescriptor) ([][]byte, error) {
 	return dlist, nil
 }
 
-func LoadSchemaRegistryFromFile(schemaReg SchemaRegistry, nsID ident.ID, protoFile string, msgName string, importPathPrefix ...string) error {
-	out, _ := parseProto(protoFile, importPathPrefix...)
+func AppendSchemaOptions(schemaOpt *nsproto.SchemaOptions, protoFile, msgName string, contents map[string]string, deployID string, importPath ...string) (*nsproto.SchemaOptions, error) {
+	sh, err := LoadSchemaHistory(schemaOpt)
+	var prevID string
+	if descr, ok := sh.GetLatest(); ok {
+		prevID = descr.DeployId()
+	}
 
-	dlist, _ := marshalFileDescriptors(out)
+	if err != nil {
```

Comment:
shouldnt this error check happen right after its returned? If you got an error, you've already tried to use the schema history

Reply:
good catch